pyClient/README.md

```diff
@@ -45,11 +55,211 @@ and fails to fetch it (or fails to fetch the right version).
 
 ```console
 # Download the solidity compiler to compile the contracts
-$ wget https://github.com/ethereum/solidity/releases/download/[solc-version]/[solc-for-your-distribution] -O $ZETH/pyClient/zeth-devenv/lib/[python-version]/site-packages/solcx/bin/solc-[solc-version]
+$ wget https://github.com/ethereum/solidity/releases/download/[solc-version]/[solc-for-your-distribution] \
+    -O $ZETH/pyClient/zeth-devenv/lib/[python-version]/site-packages/solcx/bin/solc-[solc-version]
 $ chmod +x $ZETH/pyClient/zeth-devenv/lib/[python-version]/site-packages/solcx/bin/solc-[solc-version]
 ```
 
-To run this command you need to replace the solidity version (denoted by
-`[solc-version]`), the python version (denoted `[python-version]`), and binary
-file (denoted by `[solc-for-your-distribution]`) by your system specific pieces
-of information.
+To run this command, replace the solidity version (denoted by `[solc-version]`),
+the python version (denoted by `[python-version]`), and binary file (denoted by
+`[solc-for-your-distribution]`) by your system specific information.
+
+# The `zeth` command line interface
+
+The `zeth` command exposes Zeth operations via a command line interface.  A
+brief description is given in this section.  More details are available via
+`zeth --help`, and example usage can be seen in the [pyclient test
+script](../scripts/test_zeth_cli).
+
+## Environment
+
+Depending on the operation being performed, the `zeth` client must:
+- interact with an Ethereum RPC host
+- interact with the deployed Zeth contracts
+- request proofs and keys from the `prover_server`, and
+- access secret and public data for the current user
+
+Some of this data could not easily be specified by command line, so the `zeth`
+tools assume it can be found in specific files (with default file names,
+overridable on the command line).
+
+The set of files required by zeth for a single user is described below.  We
+recommend creating a directory for each user, containing these files.  In this
+way, it is very easy to setup one or more conceptual "users" and invoke `zeth`
+operations on behalf of each of them to experiment with the system.
+
+- `eth-address` specifies an eth address from which to fund transactions.  When
+  running the testnet (as described in the [top-level README](../README.md)),
+  addresses are created at startup and written to the console.  One of these can
+  be copy-pasted into this file.
+- `zeth-instance.json` contains the address and interface for an instance of the
+  zeth contracts.  This file is created by the deployment step below and should
+  be distributed to each client.
```

Comment:
No, `zeth-instance.json` contains the info about the instance that the given client shoudl itneract with (the instance whose mekle tree contains the notes in the `notes` dir).  We can't use a single "client" (collection of config files, addresses and notes) across multiple instances.

Reply:
Rewored to try and make this clearer.

Reply:
> No, zeth-instance.json contains the info about the instance that the given client shoudl itneract with (the instance whose mekle tree contains the notes in the notes dir).

Aren't we saying the same thing here? By "Do you mean that it contains the address of all deployed Zeth contracts and their ABIs?" I meant that the ABIs of the deployed Mixer, Merkle Tree, Verifier contracts (and their respective addresses) **with which the client will interact** are all written in the `zeth-instance.json` file. In other words, the client only talks to a given deployed `Mixer` contract instance, but there are other contracts that need to be deployed on-chain, such as the MKTree, Verifier, ERC token (if applicable) contracts.

Reply:
> Hum... aren't we saying the same thing here? ...

Ah, OK.  I thought you meant something else.  
`zeth-instance.json` contains the mixer details (address and ABI) and the same for the token, if one is present.  It does not contain the data about the other contracts (verifiers, hashers etc) because AFAICT, the client does not interact with these directly (and also in the optimization PR I've turned some of these into libraries to help save gas during `mix` calls).

Reply:
Why not call the Mixer contract by its name and:
1. Rename `zeth-instance.json` to `mixer-instance.json` (containing the ABI and address of the Mixer contract deployed)
2. Rephrase to "`mixer-instance.json` contains the address and ABI of the deployed Mixer contract"

If needed, ie: if one wants to use a token, we can still have another file `token-instance.json` that is used if the `--token` flag is set to the mix function

Reply:
these `*-instance.json` files can be stored in a `instances` folder under each client's repository

I think in the end we could have something like:
```
Alice client repo
|_ Ether
|    |_ notes
|    |_ keys
|    |_ config
|        |_ mixer-instance.json
|_ ERCToken1
|    |_ notes
|    |_ keys
|    |_ config
|        |_ mixer-instance.json
|        |_ token-instance.json
|_ ERCToken2
|    |_ notes
|    |_ keys
|    |_ config
|        |_ mixer-instance.json
|        |_ token-instance.json
|_ ...
```

Where we have a mixer per asset being mixed, and where the list of asset folders represent the list of supported assets (Ether and a list of ERC20/223 tokens) by the wallet `

Reply:
@AntoineRondelet I've added a bit more to the README to try and make this clearer, including a directory structure based on your comment here.  I'm not sure if it's too much, but I thought it better to over-explain rather than leave room for confusion.  Let me know what you think.  It might make sense to reword some of it.  Thanks.

Reply:
Thanks @dtebbs, just had a look, we can keep it that way for now. I'm sure there is place for a better management of the configuration but it's good to see that we can easily support mixing multiple assets by just following some conventions on how files are organized, and running the commands in the right directory. That's more than enough for now.